### DISCUSSION TITLE
CustomDataTemplate w/UserControl does not function

### DISCUSSION THREAD

**Discussion by DBenS** [AvaloniaUI/Avalonia.Samples#84] [Q&A] [answered]
The _Basic DataTemplate Sample - Approach 3_ (Place a UserControl inside a Data Template) is not working.

## Describe the bug
When set in a project called "CustomDataTemplate", I have an error on the "view" statement of the MainWindow.axaml:
- **Unable to resolve type PersonView from namespace using:CustomDataTemplate.Views Line 82, position 6**.

- [X] Typos in the docs
- [ ] Missing information
- [ ] Misleading or wrong information
- [ ] Outdated links
- [X] Outdated or not working source code

## Sample affected
[https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/BasicDataTemplateSample#approach-3-place-a-usercontrol-inside-a-datatemplate](https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/BasicDataTemplateSample#approach-3-place-a-usercontrol-inside-a-datatemplate
)

## Screenshots
![image](https://github.com/AvaloniaUI/Avalonia.Samples/assets/13090884/adb3fe60-8f68-44c1-bfa8-988d22e71eac)

**Comment by timunie** [comment]
@DBenS it works for me without any change. Did you run it from source code or from your own project? Does the sample compile and run without an error? 

> [!TIP]
> Make sure you have setup the xml-namespace at the root node. `xmlns:view="using:BasicDataTemplateSample.Views"`

> [!NOTE]
> The editor needs to have a fresh build after adding a new View, otherwise it can't find it and then you get this error.

**Comment by timunie** [comment]
need a sample to continue

**Reply by DBenS** [reply]
I just followed the code listed in 
[https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/BasicDataTemplateSample](https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/BasicDataTemplateSample)

... and I'm having a compiling issue:
Avalonia error AVLN:0004: Unable to resolve type PersonView from namespace using:CustomDataTemplate.Views Line 82, position 6.

I'm sorry for taking so much time to answer you, and thank you for your support.

[Custom DataTemplate in a Control.zip](https://github.com/AvaloniaUI/Avalonia.Samples/files/15409765/Custom.DataTemplate.in.a.Control.zip)

**Reply by timunie** [reply, marked as answer]
Now I see. You have a mistake in your namespace for some reason. Maybe due to added PersonView in the wrong location and moved into the correct folder afterwards. Namespaces don't cover that automatically. 

In PersonView.axaml.cs
```diff
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;

- namespace CustomDataTemplate;
+ namespace CustomDataTemplate.Views;

[ ... ]
```

In PersonView.axaml
```diff
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
-            x:Class="CustomDataTemplate.PersonView"
+            x:Class="CustomDataTemplate.Views.PersonView"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"            
             xmlns:model="using:CustomDataTemplate.Models" 
             xmlns:vm="using:CustomDataTemplate.ViewModels"
             x:CompileBindings="True" 
             x:DataType="model:Person">

[ ... ]
```

**Reply by DBenS** [reply]
You are GREAT @timunie !
Thank you very much!
